src/data_lakehouse_ingest/orchestrator/schema_utils.py

```diff
@@ -0,0 +1,131 @@
+"""
+Schema management utilities for the Data Lakehouse Ingest framework.
+Handles schema resolution (LinkML or inline SQL) and column alignment for ingested DataFrames.
+Provides helpers to enforce consistent structure between raw data and curated Delta tables.
+"""
+from minio import Minio
+from pyspark.sql import SparkSession, DataFrame
+import logging
+
+def resolve_schema(
+    spark: SparkSession,
+    table: dict[str, object],
+    logger: logging.Logger,
+    minio_client: Minio | None = None,
+) -> tuple[str | None, str]:
+    """
+    Resolve the schema definition for a given table.
+
+    Determines whether to use a LinkML schema or a provided inline SQL schema
+    (`schema_sql`), falling back to an inferred schema if neither is available.
+
+    If a LinkML schema is defined, it takes precedence over `schema_sql`. In that
+    case, the schema is parsed and converted into a Spark-compatible SQL string.
+
+    Args:
+        spark (SparkSession): Active Spark session used for schema parsing.
+        table (dict): Table definition dictionary containing schema fields
+            such as "schema_sql" and/or "linkml_schema".
+        logger (logging.Logger): Logger for structured information and errors.
+        minio_client (Minio | None): Optional MinIO client used to fetch LinkML
+            schema files from S3-compatible storage.
+
+    Returns:
+        Tuple[str | None, str]: A tuple containing:
+            - schema_sql (str | None): The resolved SQL-style schema string,
+              e.g. "id INT, name STRING".
+            - schema_source (str): The source of the schema, one of:
+                {"linkml", "schema_sql", "fallback_sql", "inferred", "custom_parser"}.
+
+    Notes:
+        - LinkML schema takes precedence and is converted into SQL syntax.
+        - If LinkML parsing fails but a SQL schema exists, the function falls
+          back to "fallback_sql".
+        - If neither schema is defined, returns (None, "inferred").
+    """
+    schema_sql = table.get("schema_sql")
+    linkml_schema = table.get("linkml_schema")
+
+    if linkml_schema:
+        logger.info(
+            f"LinkML schema detected for table {table.get('name')} "
+            "(feature not implemented yet; falling back)"
+        )
+        # TODO: Implement LinkML schema parsing once linkml_parser is ready
+        if schema_sql:
+            return schema_sql, "fallback_sql"
+        else:
+            logger.warning("No schema_sql fallback available. Using inferred schema.")
+            return None, "inferred"
```

Comment:
Same comment as above really but if I submit a linkml schema my expectation is that it gets used or I get an error. My choice as a user should never be silently ignored